SportsHub/SportsHub.DAL/Repository/ArticleRepository.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Microsoft.EntityFrameworkCore;
+using SportsHub.DAL.Data;
+using SportsHub.Domain.Models;
+using SportsHub.Domain.Repository;
+
+namespace SportsHub.DAL.Repository
+{
+    public class ArticleRepository : GenericRepository<Article>, IArticleRepository
+    {
+        public ArticleRepository(ApplicationDbContext context) : base(context)
+        {
+        }
+
+        public async Task<Article?> GetByTitleAsync(string title)
+        {
+            return await DbSet
+                .Where(x => x.Title == title)
+                .FirstOrDefaultAsync();
```

Comment:
why first or default? there is a constraint that the Title is unique?

Reply:
But `FirstAsync()` will throw and exception if there is no such article.

Reply:
I understand, but my question about others: what is we have a few articles with the same title? how do you know that you need the first one?